libs/shared/base-abstract-classes/src/service/editors/multi-row-editor.service.ts

```diff
@@ -43,6 +46,25 @@ export abstract class MultiRowEditorService<T extends TableRow> extends EditorSe
     this.initForm();
   }
 
+  private getRowIndex(id: string | number): number {
+    for (let i = 0; i < this._newRows.length; i++) {
+      if (id === this._newRows[i][this._entitySecondIdField]) {
+        return i;
+      }
+    }
```

Comment:
What about using .findIndex?
If the index result is -1 then return 0, otherwise return the index that you found.

Reply:
let's refactor it in another PR (this code is not new, I just moved it), there are other little things that can be improved but I don't like to make the PR bigger than the original scope

Reply:
ok